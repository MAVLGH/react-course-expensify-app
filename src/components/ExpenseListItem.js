import React from 'react';
import { NavLink } from 'react-router-dom';
import moment from 'moment';

const ExpenseListItem = ({ description, amount, createdAt, id }) => (
    <div>
        <h3><NavLink to={`/edit/${id}`} activeClassName="is-active" exact={true}>{description}</NavLink></h3>
        <p>Amount: {`USD$${(amount / 100)} `}-{` Date: ${moment(createdAt).format('DD/MM/YYYY HH:mm:ss').toString()}`}</p>
    </div>
);

export default ExpenseListItem;

