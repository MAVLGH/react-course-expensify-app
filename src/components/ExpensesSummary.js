import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpensesTotal from '../selectors/expenses-total'
import selectExpenses from '../selectors/expenses'

export const ExpensesSummary = ({ expensesLength, expensesTotal }) => {

  const expensesTotalFormatted = `USD${numeral(expensesTotal / 100).format('$0,0.00')} `
  const expenseWord = expensesLength === 1 ? 'expense' : 'expenses'
  return (
    <div>
      <h1>Expenses Summary</h1>
      {expensesLength > 0 ? <p>Viewing {expensesLength} {expenseWord} totalling {expensesTotalFormatted}</p> : <p>No summary</p>}
    </div>
  )
}

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters)
  return {
    expensesLength: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  }
}

export default connect(mapStateToProps)(ExpensesSummary)