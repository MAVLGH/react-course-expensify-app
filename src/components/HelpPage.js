import React from 'react'

const HelpPage = () => (
    <div>
        This is from my expense Help component
    </div>
);

export default HelpPage;