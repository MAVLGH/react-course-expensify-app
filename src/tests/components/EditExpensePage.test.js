import React from 'react';
import { shallow } from 'enzyme';
import { EditExpensePage } from '../../components/EditExpensePage';
import expenses from '../fixtures/expenses'

let editExpense, removeExpense, history, wrapper;

beforeEach(() => {
  editExpense = jest.fn() // spy
  removeExpense = jest.fn() //spy
  history = { push: jest.fn() }; // spy
  wrapper = shallow(
    <EditExpensePage
      expense={expenses[0]}
      removeExpense={removeExpense}
      editExpense={editExpense}
      history={history}
    />)
})


test("should render EditExpensePage correctly", () => {
  expect(wrapper).toMatchSnapshot();
})

test("should handle editExpense", () => {
  const update = {
    description: 'edited description'
  }
  const updatedExpense = { ...expenses[0], ...update }
  wrapper.find('ExpenseForm').prop('onSubmit')(updatedExpense)
  expect(history.push).toHaveBeenLastCalledWith('/');
  expect(editExpense).toHaveBeenLastCalledWith(updatedExpense.id, updatedExpense);
})

test("should handle removeExpense", () => {
  wrapper.find('button').simulate('click')
  expect(history.push).toHaveBeenLastCalledWith('/');
  expect(removeExpense).toHaveBeenLastCalledWith(expenses[0].id);
})