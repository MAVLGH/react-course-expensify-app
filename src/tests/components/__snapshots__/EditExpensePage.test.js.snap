// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditExpensePage correctly 1`] = `
<div>
  <h3>
    Editing the expense
  </h3>
  <ExpenseForm
    expense={
      Object {
        "amount": 195,
        "createdAt": 0,
        "description": "Gum",
        "id": "0",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
